build: increase logictest stress timeout to 2h

asubiotto opened the pull request:
The default 40 minutes timeout is not enough.

Release note: None (testing change)

Closes #53729

cc @yuzefovich

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55971)
<!-- Reviewable:end -->

Commit:
build: increase logictest stress timeout to 2h

The default 40 minutes timeout is not enough.

Release note: None (testing change)

asubiotto commented:
TFTR

bors r=tbg

asubiotto commented:
Looks like a `lint` infra flake?

bors r=tbg
